specs/tinted8/builder.md

```diff
@@ -0,0 +1,308 @@
+# Tinted8 Builder Guidelines
+
+**Version 0.1.0** The latest version of this spec can be obtained from
+[tinted-theming/specs/tinted8/builder](https://github.com/tinted-theming/home/blob/main/specs/tinted8/builder.md)
+
+## Introduction
+
+Builders convert a Tinted8 scheme into color data for templates to generate
+themes.
+
+## Inputs
+
+### Scheme Inputs
+
+Builders read scheme files that conform to the Styling specification. At
+minimum they must provide:
+
+- Required: `system`, `scheme-author`, `variant`, `palette`
+- Partially required: At least one of `name`, `slug` or `family`
+- Optional: `syntax`, `ui`
+
+All color values must be HTML-style hex (`#RRGGBB`).
+
+### Template Input
+
+```
+/templates/config.yaml
+/templates/*.mustache
+```
+
+`config.yaml` describes which systems the template supports and how output
+filenames are constructed. Builders read these values to generate consistent
+output paths.
+
+## Name and Slug Handling
+
+If a scheme lacks a `slug`, builders derive one by slugifying the `name`:
+
+- Normalize Unicode to ASCII
+- Lowercase all letters
+- Replace spaces with `-`
+- Remove non-alphanumeric, non-dash characters
+
+Examples: 
+
+- `Catppuccin Mocha` → `catppuccin-mocha`
+- `Rosé Pine` → `rose-pine`
+- `Default (Dark)` → `default-dark`
+
+## Palette Expansion
+
+### Variants
+
+For every `palette.{{token_name}}`, builders generate:
+
+- `default` - The color as provided (e.g. `red-default`)
+- `bright` - A lighter variant (e.g. `cyan-bright`)
+- `dim` - A darker variant (e.g `green-dim`)
```

Comment:
How are these actually generated? This seems to be designed around specifying base colors and generating in-between colors, but "bright" and "dim" seem a bit under-defined.